core/src/main/java/org/elasticsearch/client/transport/TransportClient.java

```diff
@@ -164,7 +164,8 @@ private static ClientTemplate buildTemplate(Settings providedSettings, Settings
 
             Injector injector = modules.createInjector();
             final TransportClientNodesService nodesService =
-                new TransportClientNodesService(settings, transportService, threadPool);
+                new TransportClientNodesService(settings, transportService, threadPool, failureListner == null
+                    ? (t,e) -> {} :failureListner);
```

Comment:
Nit: Do you mind fixing the spacing here: `(t,e) -> {} :failureListner` -> `(t, e) -> {} : failureListner`?